packages/client/src/api/client.ts

```diff
@@ -21,1907 +18,72 @@ export interface XataApiClientOptions {
-    return operationsByTag.databases.createDatabase({
-      pathParams: { workspaceId: workspace, dbName: database },
-      body: data,
-      headers,
-      ...this.extraProps
-    });
-  }
-
-  public deleteDatabase({
-    workspace,
-    database
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-  }): Promise<Components.DeleteDatabaseResponse> {
-    return operationsByTag.databases.deleteDatabase({
-      pathParams: { workspaceId: workspace, dbName: database },
-      ...this.extraProps
-    });
-  }
-
-  public getDatabaseMetadata({
-    workspace,
-    database
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-  }): Promise<Schemas.DatabaseMetadata> {
-    return operationsByTag.databases.getDatabaseMetadata({
-      pathParams: { workspaceId: workspace, dbName: database },
-      ...this.extraProps
-    });
-  }
-
-  public updateDatabaseMetadata({
-    workspace,
-    database,
-    metadata
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-    metadata: Schemas.DatabaseMetadata;
-  }): Promise<Schemas.DatabaseMetadata> {
-    return operationsByTag.databases.updateDatabaseMetadata({
-      pathParams: { workspaceId: workspace, dbName: database },
-      body: metadata,
-      ...this.extraProps
-    });
-  }
-
-  public renameDatabase({
-    workspace,
-    database,
-    newName
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-    newName: Schemas.DBName;
-  }): Promise<Schemas.DatabaseMetadata> {
-    return operationsByTag.databases.renameDatabase({
-      pathParams: { workspaceId: workspace, dbName: database },
-      body: { newName },
-      ...this.extraProps
-    });
-  }
-
-  public getDatabaseGithubSettings({
-    workspace,
-    database
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-  }): Promise<Schemas.DatabaseGithubSettings> {
-    return operationsByTag.databases.getDatabaseGithubSettings({
-      pathParams: { workspaceId: workspace, dbName: database },
-      ...this.extraProps
-    });
-  }
-
-  public updateDatabaseGithubSettings({
-    workspace,
-    database,
-    settings
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-    settings: Schemas.DatabaseGithubSettings;
-  }): Promise<Schemas.DatabaseGithubSettings> {
-    return operationsByTag.databases.updateDatabaseGithubSettings({
-      pathParams: { workspaceId: workspace, dbName: database },
-      body: settings,
-      ...this.extraProps
-    });
-  }
-
-  public deleteDatabaseGithubSettings({
-    workspace,
-    database
-  }: {
-    workspace: Schemas.WorkspaceID;
-    database: Schemas.DBName;
-  }): Promise<void> {
-    return operationsByTag.databases.deleteDatabaseGithubSettings({
-      pathParams: { workspaceId: workspace, dbName: database },
-      ...this.extraProps
-    });
-  }
+      });
+    }
+  } as unknown as { new (options?: XataApiClientOptions): XataApiProxy };
```

Comment:
Is the `new` part necessary? This syntax is confusing/new to me

Reply:
Haha, yeah! This is how `class` types are defined where the `new` clause refers to the constructor of a class. As a fun fact, classes are syntactic sugar for objects with a constructor that is injected in the new property.